sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3075,7 +3085,9 @@ object SQLConf {
       RemovedConfig("spark.sql.optimizer.planChangeLog.rules", "3.1.0", "",
         s"Please use `${PLAN_CHANGE_LOG_RULES.key}` instead."),
       RemovedConfig("spark.sql.optimizer.planChangeLog.batches", "3.1.0", "",
-        s"Please use `${PLAN_CHANGE_LOG_BATCHES.key}` instead.")
+        s"Please use `${PLAN_CHANGE_LOG_BATCHES.key}` instead."),
+      RemovedConfig("spark.sql.sources.schemaStringLengthThreshold", "3.2.0", "4000",
```

Comment:
I removed the old config, because people are unlikely to set it (4000 is the actual hive limitation), and we can't fallback to static config in dynamic config (I tried and there are object initialization issues).

Reply:
Is there any case when users set `spark.sql.hive.tablePropertyLengthThreshold` to make it working then? Looks like we don't have to expose a configuration at all if it's still unlikely for people to set.

Reply:
e.g. maybe users have a hive-compatible metastore which doesn't have this limitation or have a different limitation (Glue).

It's also useful for testing :)